site/src/main/scala/org/typelevel/sbt/TypelevelSitePlugin.scala

```diff
@@ -128,30 +131,50 @@ object TypelevelSitePlugin extends AutoPlugin {
         name = Some("Generate site")
       )
     ),
-    tlSitePublish := List(
-      WorkflowStep.Use(
-        UseRef.Public("peaceiris", "actions-gh-pages", "v3.8.0"),
-        Map(
-          "github_token" -> s"$${{ secrets.GITHUB_TOKEN }}",
-          "publish_dir" -> (ThisBuild / baseDirectory)
-            .value
-            .toPath
-            .toAbsolutePath
-            .relativize((laikaSite / target).value.toPath)
-            .toString,
-          "keep_files" -> tlSiteKeepFiles.value.toString
-        ),
-        name = Some("Publish site"),
-        cond = {
-          val predicate = tlSitePublishBranch
-            .value // Either publish from branch or on tags, not both
-            .fold[RefPredicate](RefPredicate.StartsWith(Ref.Tag("v")))(b =>
-              RefPredicate.Equals(Ref.Branch(b)))
-          val publicationCond = GenerativePlugin.compileBranchPredicate("github.ref", predicate)
-          Some(s"github.event_name != 'pull_request' && $publicationCond")
-        }
-      )
-    ),
+    tlSitePublish := {
+      def publishSiteWorkflowStep(publishPredicate: RefPredicate) =
+        List(
+          WorkflowStep.Use(
+            UseRef.Public("peaceiris", "actions-gh-pages", "v3.8.0"),
+            Map(
+              "github_token" -> s"$${{ secrets.GITHUB_TOKEN }}",
+              "publish_dir" -> (ThisBuild / baseDirectory)
+                .value
+                .toPath
+                .toAbsolutePath
+                .relativize((laikaSite / target).value.toPath)
+                .toString,
+              "keep_files" -> tlSiteKeepFiles.value.toString
+            ),
+            name = Some("Publish site"),
+            cond = {
+              val predicate = publishPredicate
+              val publicationCond =
+                GenerativePlugin.compileBranchPredicate("github.ref", predicate)
+              Some(s"github.event_name != 'pull_request' && $publicationCond")
+            }
+          )
+        )
+
+      val tlSitePublishTagsV = tlSitePublishTags.value
+      val tlSitePublishBranchV = tlSitePublishBranch.value
+
+      tlSitePublishTagsV -> tlSitePublishBranchV match {
```

Comment:
Nit :) I only use `->` for key-value pairs.

```suggestion
      (tlSitePublishTagsV, tlSitePublishBranchV) match {
```

Reply:
So be it.